addons/web/static/src/core/domain_selector/fields/domain_selector_binary_field.js

```diff
@@ -0,0 +1,26 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { DomainSelectorFieldInput } from "./domain_selector_field_input";
+
+import { Component } from "@odoo/owl";
+
+const dsf = registry.category("domain_selector/fields");
+const dso = registry.category("domain_selector/operator");
+
+export class DomainSelectorBinaryField extends Component {}
+Object.assign(DomainSelectorBinaryField, {
+    template: "web.DomainSelectorBinaryField",
+    components: {
+        DomainSelectorFieldInput,
+    },
+
+    onDidTypeChange() {
+        return { value: "" };
+    },
+    getOperators() {
+        return ["=", "!=", "set", "not set"].map((key) => dso.get(key));
+    },
+});
+
+dsf.add("binary", DomainSelectorBinaryField);
```

Comment:
don't add to registry but use it here https://github.com/odoo/odoo/blob/1d630e9261a750838923d801bf1d9d3316353529/addons/web/static/src/core/domain_selector/domain_selector_leaf_node.js#L49